android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java

```diff
@@ -184,6 +184,9 @@ private void restoreCache() {
       JSONObject serializedCollection = new JSONObject(prefs.getString(PW_COLLECTION_KEY, null));
       mPwCollection = PhysicalWebCollection.jsonDeserialize(serializedCollection);
       Utils.setPwsEndpoint(this, mPwCollection);
+      for (UrlDevice urlDevice : mPwCollection.getDeviceIdToUrlDeviceMap().values()) {
+        Utils.updateRegion(urlDevice);
```

Comment:
You need to make sure the URL device has rssi and txpower before doing this (check isBleUrlDevice)